pkg/apis/logging/v1/cluster_log_forwarder_types.go

```diff
@@ -211,6 +216,26 @@ type PipelineSpec struct {
 	Parse string `json:"parse,omitempty"`
 }
 
+type OutputDefaults struct {
+
+	// Type of output for which default values need to be set
+	//
+	// +kubebuilder:validation:Enum:=default
+	// +required
+	Type string `json:"type,omitempty"`
+
+	// StructuredIndexKey specifies the metadata key to be used as name of elasticsearch index
+	// It takes precedence over IndexName
+	//
+	// +optional
+	StructuredIndexKey string `json:"structuredIndexKey,omitempty"`
+
+	// StructuredIndexName specifies the name of elasticsearch schema
+	//
+	// +optional
+	StructuredIndexName string `json:"structuredIndexName,omitempty"`
+}
+
```

Comment:
Very sorry about this, it's my fault. You followed the proposal, I wrote it wrong. Here's what I meant to write:

```suggestion
type OutputDefaults struct {
	// +optional
	Elasticsearch *Elasticsearch `json:"elasticsearch,omitempty"`
}
```
This way we re-use the existing ES settings struct, and it's easy to extend to cover other output types in future if we need to. I'll add a JIRA to cover other output types later - ultimately OutputDefaults == OutputTypeSpec so we automatically allow default settings for any new output type that is added.

Reply:
updated